source/slang/slang-stdlib-textures.cpp

```diff
@@ -0,0 +1,1213 @@
+
+static String cudaSampleIntrinsic(const bool isArray, const BaseTextureShapeInfo& base, bool sampleLevel)
+{
+    StringBuilder cudaBuilder;
+
+    TextureFlavor::Shape baseShape = base.baseShape;
+    const int coordCount = base.coordCount;
+    const int vecCount = coordCount + int(isArray);
+
+    if( baseShape != TextureFlavor::Shape::ShapeCube )
+    {
+        cudaBuilder << "tex" << coordCount << "D";
+        if (isArray)
+            cudaBuilder << "Layered";
+        if(sampleLevel)
+            cudaBuilder << "Lod";
+        cudaBuilder << "<$T0>($0";
+        for (int i = 0; i < coordCount; ++i)
+        {
+            cudaBuilder << ", ($2)";
+            if (vecCount > 1)
+            {
+                cudaBuilder << '.' << char(i + 'x');
+            }
+        }
+        if (isArray)
+            cudaBuilder << ", int(($2)." << char(coordCount + 'x') << ")";
+        if(sampleLevel)
+            cudaBuilder << ", $3";
+        cudaBuilder << ")";
+    }
+    else
+    {
+        cudaBuilder << "texCubemap";
+        if (isArray)
+            cudaBuilder << "Layered";
+        if(sampleLevel)
+            cudaBuilder << "Lod";
+        cudaBuilder << "<$T0>($0, ($2).x, ($2).y, ($2).z";
+        if (isArray)
+            cudaBuilder << ", int(($2).w)";
+        if(sampleLevel)
+            cudaBuilder << ", $3";
+        cudaBuilder << ")";
+    }
+
+    return cudaBuilder;
+}
+
+const char* noBias = nullptr;
+const char* noLodLevel = nullptr;
+const char* noGradX = nullptr;
+const char* noGradY = nullptr;
+const char* noConstOffset = nullptr;
+const char* noMinLod = nullptr;
+
+static String spirvSampleIntrinsic(
+    const TextureTypePrefixInfo& prefixInfo,
+    const char* bias = nullptr,
+    const char* lodLevel = nullptr,
+    const char* gradX = nullptr,
+    const char* gradY = nullptr,
+    const char* constOffset = nullptr,
+    const char* minLod = nullptr)
+{
+    StringBuilder spirvBuilder;
+    const char* i = "                ";
+
+    SLANG_ASSERT(!(!gradX ^ !gradY));
+
+    if(minLod)
+        spirvBuilder << i << "OpCapability MinLod;\n";
+
+    const char* sampledImage;
+    if(prefixInfo.combined)
+    {
+        sampledImage = "$this";
+    }
+    else
+    {
+        const char* sampledImageType = "%sampledImageType";
+        sampledImage = "%sampledImage";
+        spirvBuilder << i << sampledImageType << " = OpTypeSampledImage $$This;\n";
+        spirvBuilder << i << sampledImage << " : " << sampledImageType << " = OpSampledImage $this $s;\n";
+    }
+
+    const char* op = lodLevel || gradX ? "OpImageSampleExplicitLod" : "OpImageSampleImplicitLod";
+    spirvBuilder << i << "%sampled : __sampledType(T) = " << op << " " << sampledImage << " $location";
+    spirvBuilder << " None";
+    if(bias)
+        spirvBuilder << "|Bias";
+    if(lodLevel)
+        spirvBuilder << "|Lod";
+    if(gradX)
+        spirvBuilder << "|Grad";
+    if(constOffset)
+        spirvBuilder << "|ConstOffset";
+    if(minLod)
```

Comment:
Also need `Sample` before `minlod`.

Reply:
Yeah, I think this bit will get cleaned up, it was just implemented lazily so far.